rabbitmq_polycymaker/rabbitmq_policy.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+
+import logging
+from dataclasses import dataclass
+from hashlib import sha1
+from re import escape
+from time import sleep
+from typing import Dict, List
+
+from pyrabbit2 import Client
+
+log = logging.getLogger()
+
+RUNNING = "running"
+
+
+def bucket(string, size):
+    hs = int(sha1(string.encode("utf-8")).hexdigest(), 16)
+    return hs % size
+
+
+@dataclass
+class QueueWithoutPolicy:
+    vhost: str
+    name: str
+
+
+class RabbitData:
+    def __init__(
+        self,
+        client: Client,
+        policy_groups: Dict,
+        dry_run: bool,
+        wait_sleep: int,
+    ):
+        self.client = client
+        self.policy_groups = policy_groups
+        self.dry_run = dry_run
+        self.wait_sleep = wait_sleep
+
+    def reload(self):
+        self.client.get_vhost_names()
+        self.client.get_queues()
+        self.client.get_nodes()
+
+    def queues_without_policy(self) -> List[QueueWithoutPolicy]:
+
+        queues_list = []
+
+        for queue in self.client.get_queues():
+            exclusive = queue.get("exclusive")
+            auto_delete = queue.get("auto_delete")
+
+            if exclusive or auto_delete:
+                continue
+
+            name = queue.get("name")
+            vhost = queue.get("vhost")
+            policy = queue.get("policy")
+            if not policy or name != policy:
+                queues_list.append(QueueWithoutPolicy(vhost=vhost, name=name))
+
+        return queues_list
+
+    def is_queue_running(self, vhost: str, queue: str) -> bool:
+        state = None
+        while state != RUNNING:
+            try:
+                state = self.client.get_queue(vhost, queue)["state"]
+                log.info("Queue %r has state %r", queue, state)
+                if state != RUNNING:
+                    sleep(self.wait_sleep)
+                else:
+                    return True
+            except KeyError:
+                log.exception("RabbitMQ API not ready to answer")
+                sleep(self.wait_sleep)
+
+    def create_policy(self, vhost: str, queue: str):
+
+        bucket_number = bucket(
+            "{}{}".format(vhost, queue), len(self.policy_groups)
+        )
+        bucket_nodes = self.policy_groups.get(str(bucket_number))
+
+        rabbit_nodes = []
+
+        for node in bucket_nodes:
+            rabbit_nodes.append("rabbit@{}".format(node))
```

Comment:
А нужен ли 3.6? Всё-таки `f"rabbit@{node}"` смотрелось бы значительно лучше.
Does 3.6 really necessary? It will much nicer to have `f"rabbit@{node}"` here.